File: .idea/app/src/main/java/com/chevstrap/rbx/Paths.java
package com.chevstrap.rbx;

import java.io.File;

public class Paths {
    private static String base = "";
    private static String backups = "";
    private static String logs = "";
    private static String modifications = "";

    public static String getTemp() {
        return new File(App.getAppContext().getCacheDir(), "").getAbsolutePath();
    }

    public static String getLocalAppData() {
        return App.getAppContext().getFilesDir().getAbsolutePath();
    }

    public static String getBackups() {
        ensureInitialized();
        return backups;
    }

    public static String getLogs() {
        ensureInitialized();
        return logs;
    }

    public static String getModifications() {
        ensureInitialized();
        return modifications;
    }

    public static boolean isInitialized() {
        return !base.isEmpty();
    }

    public static void initialize(String baseDirectory) {
        base = baseDirectory;
        backups = new File(base, "Backups").getAbsolutePath();
        logs = new File(base, "Logs").getAbsolutePath();
        modifications = new File(base, "Modifications").getAbsolutePath();
    }

    private static void ensureInitialized() {
        if (!isInitialized()) {
            initialize(getLocalAppData());
        }
    }
}

File: app/src/main/java/com/chevstrap/rbx/Paths.kt
package com.chevstrap.rbx

import java.io.File

object Paths {
    private var base = ""
    private var logsDir = ""
    private var customBackgroundDir = ""
    private var modificationsDir = ""

    @JvmStatic
    val localAppData: String
        get() = App.appContext?.filesDir?.absolutePath ?: ""

    @JvmStatic
    val logs: String
        get() {
            ensureInitialized()
            return logsDir
        }

    @JvmStatic
    val customBackground: String
        get() {
            ensureInitialized()
            return customBackgroundDir
        }

    @JvmStatic
    val modifications: String
        get() {
            ensureInitialized()
            return modificationsDir
        }

    val isInitialized: Boolean
        get() = base.isNotEmpty()

    @JvmStatic
    fun initialize(baseDirectory: String) {
        base = baseDirectory
        logsDir = File(base, "Logs").absolutePath
        customBackgroundDir = File(base, "CustomBackground").absolutePath
        modificationsDir = File(base, "Modifications").absolutePath
    }

    private fun ensureInitialized() {
        if (!isInitialized) {
            initialize(localAppData)
        }
    }
}